tests/std/tests/P0896R4_ranges_subrange/test.compile.pass.cpp

```diff
@@ -23,6 +23,12 @@ using std::output_iterator_tag, std::input_iterator_tag, std::forward_iterator_t
 
 int main() {} // COMPILE-ONLY
 
+void test_LWG_3470() {
+    int a[3]                = {1, 2, 3};
+    int* b[3]               = {&a[2], &a[0], &a[1]};
+    [[maybe_unused]] auto c = std::ranges::subrange<const int* const*>(b);

```

Comment:
Could this work with CTAD?
```suggestion
    [[maybe_unused]] auto c = std::ranges::subrange{b};
```

Reply:
I think it has a purpose.

From LWG 3470:
```
The construction of c is ill-formed because convertible-to-non-slicing<int**, const int*const*> is false, although the conversion does not involve object slicing. 
```
So we need for the example that b is `int**`, without const modificators.
and we need second template argument as  `const int*const*`.
Because we test `convertible-to-non-slicing` with `int**` and ` const int*const*`, and it's working now.

I agree that we could infer the size.